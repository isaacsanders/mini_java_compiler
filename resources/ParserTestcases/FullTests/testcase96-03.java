class ElizabethDole {
    public static void main (String[] ID){
    }
}

class BobDole extends ElizabethDole {}

class Shape {}
class Ellipse extends Shape {}
class Circle extends Ellipse {} // or not
class ConvexShape extends Shape {}
class Quadrilateral extends ConvexShape {}
class Rectangle extends Quadrilateral {}
class Square extends Rectangle {}
class Rube extends Square {}
class Mark extends Rube {
    int cash;
}