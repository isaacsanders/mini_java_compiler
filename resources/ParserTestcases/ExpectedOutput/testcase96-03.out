StmtList ::= 
MainClassDecl ::= ...
ClassDeclList ::= 
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= 
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
Type ::= INT
ClassVarDecl ::= Type ID SEMI
ClassVarDeclList ::= ClassVarDeclList ClassVarDecl
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
Program ::= MainClassDecl ClassDeclList
